src/handler/mod.rs

```diff
@@ -331,13 +340,13 @@ impl Handler {
                 Some(inbound_packet) = self.socket.recv.recv() => {
                     self.process_inbound_packet::<P>(inbound_packet).await;
                 }
-                Some(Ok((node_address, pending_request))) = self.active_requests.next() => {
-                    self.handle_request_timeout(node_address, pending_request).await;
+                Some(Ok((node_address, active_request))) = self.active_requests.next() => {
+                    self.handle_request_timeout(node_address, active_request).await;
                 }
                 Some(Ok((node_address, _challenge))) = self.active_challenges.next() => {
-                    // A challenge has expired. There could be pending requests awaiting this
-                    // challenge. We process them here
-                    self.send_next_request::<P>(node_address).await;
+                    // A challenge has expired. There could be active requests impacted by this
+                    // challenge. We replay them here
+                    self.replay_active_requests::<P>(&node_address).await;
```

Comment:
I think we need to send pending_requests here not active_requests.

Current implementation, requests in pending_requests remain without being sent in case of WHOAREYOU timeout:

```mermaid
sequenceDiagram
    participant Node1
    participant Node2


    Node2 ->> Node1: Random packet
    Note over Node1: active_challenges.insert()
    Node1 ->> Node2: WHOAREYOU
    rect rgb(100, 100, 0)
    Note over Node2: ** Discard the WHOAREYOU packet<br>to reproduce kind of network issue. **
    end

    rect rgb(10, 10, 10)
    Note left of Node1: Node1 want to send FINDNODE to Node2<br>but active_challenge exists.<br>So insert requests into pending_requests.
    par 
    Note over Node1: pending_requests.insert()
    and 
    Note over Node1: pending_requests.insert()
    end
    end

    Note over Node1: The challenge in active_challenges<br>has been expired.

    rect rgb(100, 0, 0)
    Note over Node1: FINDNODE requests remain in pending_requests without being sent.
    end
```